### PR TITLE
Resolves ParticleAcceleratorPartVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the ParticleAcceleratorPartVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+using Content.Shared.Singularity.Components;
+
+namespace Content.Client.ParticleAccelerator;
+
+[RegisterComponent]
+[Access(typeof(ParticleAcceleratorPartVisualizerSystem))]
+public sealed class ParticleAcceleratorPartVisualizerComponent : Component
```
Comment (AJCM-git) [inline Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs:7]: visuals

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{
+    [DataField("baseState", required: true)]
+    public string BaseState = default!;
+
+    public static readonly Dictionary<ParticleAcceleratorVisualState, string> StatesSuffixes = new()
```
Comment (AJCM-git) [inline Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs:12]: VV

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+using Robust.Client.GameObjects;
+
+namespace Content.Client.ParticleAccelerator;
+
+public sealed class ParticleAcceleratorPartVisualizerSystem : VisualizerSystem<ParticleAcceleratorPartVisualizerComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ParticleAcceleratorPartVisualizerComponent, ComponentInit>(OnInit);
```
Comment (AJCM-git) [inline Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs:12]: oncomponentinit

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        if(!TryComp<SpriteComponent>(uid, out var sprite))
+        {
+            throw new EntityCreationException("No sprite component found in entity that has ParticleAcceleratorPartVisualizer");
+        }
+
+        if(!sprite.AllLayers.Any())
+        {
+            throw new EntityCreationException("No Layer set for entity that has ParticleAcceleratorPartVisualizer");
+        }
+    }
```
Comment (AJCM-git) [inline Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs:26]: i dont think this is necessary

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    protected override void OnAppearanceChange(EntityUid uid, ParticleAcceleratorPartVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        
+        if (!AppearanceSystem.TryGetData(uid, ParticleAcceleratorVisuals.VisualState, out ParticleAcceleratorVisualState state, args.Component))
```
Comment (AJCM-git) [inline Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs:33]: generic

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs (hunk 4 above)
Comment (TemporalOroboros) [reply Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs:26]: Alright.

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs (hunk 1 above)
Comment (TemporalOroboros) [reply Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs:7]: Done

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs (hunk 5 above)
Comment (TemporalOroboros) [reply Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs:33]: Done

**Comment by TemporalOroboros** [comment]
@DrSmugleaf 
Commits tested and pushed, reopen at your leisure.

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs (hunk 3 above)
Comment (TemporalOroboros) [reply Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs:12]: Removed as unnecessary.

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+[Access(typeof(ParticleAcceleratorPartVisualizerSystem))]
+public sealed class ParticleAcceleratorPartVisualsComponent : Component
+{
+    [DataField("stateBase", required: true)]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public string StateBase = default!;
+
+    [DataField("stateSuffixes")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public static readonly Dictionary<ParticleAcceleratorVisualState, string> StatesSuffixes = new()
```
Comment (AJCM-git) [inline Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs:15]: dont make this static

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+using Robust.Client.GameObjects;
+
+namespace Content.Client.ParticleAccelerator;
+
+public sealed class ParticleAcceleratorPartVisualizerSystem : VisualizerSystem<ParticleAcceleratorPartVisualsComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
```
[the remark this answers is not in the document]
Comment (TemporalOroboros) [reply Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerSystem.cs]: Point.

File: Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs (hunk 6 above)
Comment (TemporalOroboros) [reply Content.Client/ParticleAccelerator/ParticleAcceleratorPartVisualizerComponent.cs]: Alright.

**Pull request merged**